inventory_management_system_api/services/catalogue_item.py

```diff
@@ -209,7 +210,7 @@ def update(self, catalogue_item_id: str, catalogue_item: CatalogueItemPatchSchem
 
         return self._catalogue_item_repository.update(
             catalogue_item_id,
-            CatalogueItemIn(**{**stored_catalogue_item.model_dump(), **update_data}),
+            CatalogueItemIn(**{**stored_catalogue_item.model_dump(), **update_data, "number_of_spares": None}),
```

Comment:
```suggestion
            CatalogueItemIn(**{"number_of_spares": None, **stored_catalogue_item.model_dump(), **update_data}),
```
This prevents any existing `number_of_spares` data from being reset to `None`. However, I don't know if you want to invalidate any spares calculations after you update Catalogue item data. In that case this change isn't necessary.

Reply:
Yeah not sure why I even put it there, it should be unchanged as its the items that cause the recalculation 😄 good spot.